pkg/runtime/wfengine/workflow.go

```diff
@@ -184,20 +196,62 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 		}
 	}
 
-	// We block (re)creation of existing workflows unless they are in a completed state
-	// Or if they still have any pending activity result awaited.
-	if !created {
-		runtimeState := getRuntimeState(actorID, state)
-		if !runtimeState.IsCompleted() {
-			return fmt.Errorf("an active workflow with ID '%s' already exists", actorID)
-		}
-		if wf.activityResultAwaited.Load() {
-			return fmt.Errorf("a terminated workflow with ID '%s' is already awaiting an activity result", actorID)
+	// orchestration didn't exist and just create it.
+	if created {
+		return wf.scheduleWorkflowStart(ctx, actorID, startEvent, state)
+	}
+
+	// orchestration already exists, apply reuse id policy
+	runtimeState := getRuntimeState(actorID, state)
+	runtimeStatus := runtimeState.RuntimeStatus()
+	// if target status doesn't match, fall back to original logic, create instance only if previous one is completed
+	if !isStatusMatch(reuseIDPolicy.GetOperationStatus(), runtimeStatus) {
+		return wf.createIfCompleted(ctx, runtimeState, actorID, state, startEvent)
+	}
+
+	switch reuseIDPolicy.GetAction() {
+	case api.REUSE_ID_ACTION_IGNORE:
+		// Log an warning message and ignore creating new instance
+		wfLogger.Warnf("An instance with ID '%s' already exists; dropping duplicate create request", actorID)
```

Comment:
For consistency, we should follow the logging format that's used elsewhere in this file. For example:
```suggestion
		wfLogger.Warnf("Workflow actor '%s': ignoring request to recreate the current workflow instance", actorID)
```